crates/codegen/parser/runtime/src/cursor.rs

```diff
@@ -344,7 +341,7 @@ impl Cursor {
 
         if self.current.child_number > 0 {
             if let Some(parent_path_element) = self.path.last() {
-                let new_child_number = self.current.child_number + 1;
+                let new_child_number = self.current.child_number - 1;
```

Comment:
Ooh, good catch! It'd be great to add some simple tests to catch stuff like this at some point.